source/elements/oneDAL/include/oneapi/dal/kmeans_init.hpp

```diff
@@ -0,0 +1,97 @@
+#include "oneapi/dal/table.hpp"
+#include "oneapi/dal/train.hpp"
+
+namespace oneapi::dal::kmeans_init {
+
+namespace method {
+   /// Tag-type that denotes `dense <kmeans_init_c_math_dense_>`_
+   /// computational method.
+   struct dense {};
+
+   using by_default = dense;
+} // namespace method
+
+/// @tparam Float  The floating-point type that the algorithm uses for
+///                intermediate computations. Can be :expr:`float` or
+///                :expr:`double`.
+/// @tparam Method – Tag-type that specifies an implementation
+/// of K-Means Initialization algorithm.
+template <typename Float = float,
+          typename Method = method::by_default>
+class descriptor {
+public:
+
+   /// Creates a new instance of the descriptor with the default attribute
+   /// values.
+   explicit descriptor(std::int64_t cluster_count = 2);
+
+   /// The number of clusters $k$
+   /// @invariant :expr:`cluster_count > 0`
+   /// @remark default = 2
+   std::int64_t get_cluster_count() const;
+   descriptor& set_cluster_count(std::int64_t);
+
+};
+
+class compute_input {
+public:
+
+   /// Creates input for the computing operation with the given data,
+   /// the other attributes get default values.
+   compute_input(const table& data = table{});
+
+   /// The training set $X$
+   /// @remark default = table{}
+   const table& get_data() const;
+   compute_input& set_data(const table&);
+};
+
+class compute_result {
+public:
+
+   /// Creates a new instance of the class with the default property values.
+   compute_result();
+
+   /// The initial centroids
+   /// @remark default = table{}
+   const table& get_centroids() const;
```

Comment:
We have to specify layout of the table-how the tables stores centroids - use k-means description as the example

Reply:
fixed also for input too.